SU-SOE/stanford_soe_helper
# Merge pull request #137 from SU-SOE/SOE-2510-facebook-style

# NOT A MERGE
- This code has already been merged into the 7.x-2.x branch, however, we're trying to release a hotfix without additional commits that have been made to 7.x-2.x.  I am assuming we would tag this branch, not 7.x-2.x.  But as I have not done this before, let me know if this is the right process for such a hotfix.

# Summary
- Hotfix for facebook share images not appearing reliably.  SOE plans to send an email newsletter tomorrow morning and would like the feature working by then.  And we're trying to work around code that has been committed to the 7.x-2.x branch, but will not be part of this hotfix.

# Needed By (12/18)

# Urgency
- Perhaps a 7?

# Steps to Test
- This is a release branch.  So the steps to test might be to verify that the process described below is how we should proceed and that the release template notes look accurate and appropriate.

To create this release, I did:
```
git checkout 7.x-2.0-beta1
git checkout -b SOE-2510-2.0beta2-release
git cherry-pick -m 1 b3058e76437faf7874851390c6d311cc9f0991b0
```

### What
- Bug fix for magazine article images, so that Facebook can reliably find and share the featured image.

### So what
- Previously, Facebook was not able to find and cache the featured image, we suspect the image size was larger than Facebook's "recommended" (ie. required) horizontal image size.

### Now what
- Deploying this change should simply require a cache clear.  If there are any magazine articles with values in the Metatag module fields, we'll need to clear those values out for Facebook to find the appropriate image style.

### New Requirements and Risks (Changelog)
- This release has a new requirement on stanford_image_styles and the upcoming 7.x-4.3-alpha1 release.

### Diff
- https://github.com/SU-SOE/stanford_soe_helper/compare/582977d872e08b2f988e5f75d03850bf780a3dba...2a40ad7f47c6ffee7f82c6c23b9da23950ce0516

# Affected Projects or Products
- SOE's email newsletter.

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/SOE-2510
- https://github.com/SU-SWS/stanford_image_styles/pull/43
